azurerm/resource_arm_container_registry.go

```diff
@@ -227,8 +276,93 @@ func resourceArmContainerRegistryUpdate(d *schema.ResourceData, meta interface{}
 	return resourceArmContainerRegistryRead(d, meta)
 }
 
+func applyGeoReplicationLocations(meta interface{}, resourceGroup string, name string, oldGeoReplicationLocations []interface{}, newGeoReplicationLocations []interface{}) error {
+	replicationClient := meta.(*ArmClient).containerRegistryReplicationsClient
+	ctx := meta.(*ArmClient).StopContext
+	log.Printf("[INFO] preparing to apply geo-replications for AzureRM Container Registry.")
+
+	replicationLocationsToCreate := getReplicationLocationsToCreate(oldGeoReplicationLocations, newGeoReplicationLocations)
```

Comment:
I am not sure about what should be pulled out?

Reply:
This function isn't used anywhere else so i think its code should be put here.

Also, i think this could all be vastly simplified:
```go
createLocations := make(map[string]bool)
	for _, nl := range newLocations {
		createLocations[nl] = false
	}
	for _, ol := range oldLocations {
		if _, ok := createLocations[ol]; ok {
			createLocations[ol] = true
		}
	}
	
	for _, locationToCreate := range createLocations {
		if ! createLocations[locationToCreate]  {
			continue
		}
		//create location
	}

	// delete not needed geo-replication locations
	for _, locationToDelete := range replicationLocationsToDelete {
		if _, ok := createLocations[locationToDelete]; !ok  {
			continue
		}
		// delete locatio
	}
```

what do you think?

Reply:
Hey @katbyte - I've been able to refactor the update as you've suggested. But I've got a difficulty with using a TypeSet for location, so I rollback to TypeList. 

When using a Set, and updating the location from "eastus","westus" to "eastus","westeurope", the eastus region is not considered as a new value (which make sense in a way) but this behavior does not allow to do a diff between old and new locations:
![image](https://user-images.githubusercontent.com/4435138/48268448-49f51c00-e435-11e8-8a8f-435efe321811.png)

When using a list, I don't have these issue and I am able to diff:

![image](https://user-images.githubusercontent.com/4435138/48268472-5da08280-e435-11e8-9d71-1f6f0c372ee9.png)

I am relatively new to Terraform, so maybe there is another option I can explore?